docs/crds/routebackendextension.md

```diff
@@ -0,0 +1,243 @@
+
+##### Health Monitoring Configuration
+
+
+
+##### Backend TLS/SSL Configuration
+
+
+
+#### Usage Examples
+
+##### Basic Load Balancing Configuration
+
+```yaml
+apiVersion: ako.vmware.com/v1alpha1
+kind: RouteBackendExtension
+metadata:
+  name: basic-lb-config
+  namespace: production
+spec:
+  lbAlgorithm: LB_ALGORITHM_ROUND_ROBIN
+```
+
+##### Consistent Hash with Custom Header
+
+```yaml
+apiVersion: ako.vmware.com/v1alpha1
+kind: RouteBackendExtension
+metadata:
+  name: consistent-hash-config
+  namespace: production
+spec:
+  lbAlgorithm: LB_ALGORITHM_CONSISTENT_HASH
+  lbAlgorithmHash: LB_ALGORITHM_CONSISTENT_HASH_CUSTOM_HEADER
+  lbAlgorithmConsistentHashHdr: "X-User-ID"
+  persistenceProfile: System-Persistence-Client-IP
+```
+
+##### Health Monitoring Configuration
+
+```yaml
+apiVersion: ako.vmware.com/v1alpha1
+kind: RouteBackendExtension
+metadata:
+  name: health-monitor-config
+  namespace: production
+spec:
+  lbAlgorithm: LB_ALGORITHM_LEAST_CONNECTIONS
+  healthMonitor:
+  - kind: "AVIREF"
+    name: "http-health-check"
+  - kind: "AVIREF"
+    name: "tcp-health-check"
+```
+
+##### Backend TLS Configuration
+
+```yaml
+apiVersion: ako.vmware.com/v1alpha1
+kind: RouteBackendExtension
+metadata:
+  name: backend-tls-config
+  namespace: production
+spec:
+  lbAlgorithm: LB_ALGORITHM_LEAST_CONNECTIONS
+  backendTLS:
+    pkiProfile:
+      kind: "CRD"
+      name: "backend-pki-profile"
+    hostCheckEnabled: true
+    domainName:
+    - "api.backend.com"
+    - "service.backend.com"
+```
+
+##### Complete Configuration Example
+
+```yaml
+apiVersion: ako.vmware.com/v1alpha1
+kind: RouteBackendExtension
+metadata:
+  name: complete-backend-config
+  namespace: production
+spec:
+  # Load balancing with consistent hashing
+  lbAlgorithm: LB_ALGORITHM_CONSISTENT_HASH
+  lbAlgorithmHash: LB_ALGORITHM_CONSISTENT_HASH_SOURCE_IP_ADDRESS
+  
+  # Session persistence
+  persistenceProfile: System-Persistence-Client-IP
+  
+  # Health monitoring
+  healthMonitor:
+  - kind: "AVIREF"
+    name: "production-health-monitor"
+  
+  # Secure backend communication
+  backendTLS:
+    pkiProfile:
+      kind: "CRD"
+      name: "production-pki-profile"
+    hostCheckEnabled: true
+    domainName:
+    - "backend.production.com"
+```
+
+#### Validation Rules and Constraints
+
+The RouteBackendExtension CRD enforces the following validation rules:
+
+1. **Consistent Hash Validation**: When `lbAlgorithm` is set to `LB_ALGORITHM_CONSISTENT_HASH`, the `lbAlgorithmHash` field must be specified.
+
+2. **Custom Header Validation**: When `lbAlgorithmHash` is set to `LB_ALGORITHM_CONSISTENT_HASH_CUSTOM_HEADER`, the `lbAlgorithmConsistentHashHdr` field must be provided.
+
+3. **Domain Name Validation**: The `domainName` field in `backendTLS` can only be configured when `hostCheckEnabled` is set to `true`.
+
+4. **Namespace Scope**: All referenced objects (health monitors, PKI profiles) must be in the same namespace as the RouteBackendExtension object.
```

Comment:
All referenced objects (health monitors, PKI profiles) -> All referenced objects (PKI profiles). Since we do not use health monitor crds within RBE